### PR TITLE
cronjob: return immediately when failed to create job for the namespace is terminating

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind bug
/kind cleanup

#### What this PR does / why we need it:

1. if the namespace is being terminated, we don't have to do anything because any creation will fail.
2. do some cleanup when looking into the code here.
- -  `UpdateJob()` and `PatchJob` are not used
- - `(f *fakeJobControl) Clear()` is not used and is not a function of the interface.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/controller/cronjob/cronjob_controllerv2.go [hunk 1]
```diff
@@ -601,6 +601,9 @@ func (jm *ControllerV2) syncCronJob(
 	jobResp, err := jm.jobControl.CreateJob(cronJob.Namespace, jobReq)
 	switch {
 	case errors.HasStatusCause(err, corev1.NamespaceTerminatingCause):
+		// if the namespace is being terminated, we don't have to do
+		// anything because any creation will fail
+		return cronJob, nil, updateStatus, err
```
Comment (pacoxu) [inline pkg/controller/cronjob/cronjob_controllerv2.go:606]: The key change is here. Below are some cleanup.

File: pkg/controller/cronjob/cronjob_controllerv2_test.go [hunk 2]
```diff
@@ -1165,6 +1166,25 @@ func TestControllerV2SyncCronJob(t *testing.T) {
...
+					Field:   "metadata.namespace",
+				}}}}},
+			concurrencyPolicy:          "Allow",
+			schedule:                   onTheHour,
+			deadline:                   noDead,
+			ranPreviously:              true,
+			stillActive:                true,
+			jobCreationTime:            justAfterThePriorHour(),
+			now:                        *justAfterTheHour(),
+			expectActive:               0,
```
Comment (atiratree) [inline pkg/controller/cronjob/cronjob_controllerv2_test.go:1183]: `expectErr:  true` would be good to have

Comment (pacoxu) [reply pkg/controller/cronjob/cronjob_controllerv2_test.go:1183]: Added. Thanks for review.

**Comment by dims** [comment]
/lgtm
@soltysh this is ready for approval

**Comment by pacoxu** [comment]
ack @soltysh

**Comment by pacoxu** [comment]
ping before code freeze 
@janetkuo @soltysh

**Review by janetkuo — APPROVED** [review]
/approve

**Pull request merged**